UI/pre-stream-wizard/encoder-settings-provider-facebook.cpp

```diff
@@ -0,0 +1,244 @@
+			QTStr("PreLiveWizard.Configure.UrlError"),
+			requestUrl.toDisplayString());
+	}
+}
+
+void FacebookEncoderSettingsProvider::makeRequest(QUrl &url)
+{
+	blog(LOG_INFO, "FacebookEncoderSettingsProvider creating request");
+	QNetworkRequest request(url);
+	request.setHeader(QNetworkRequest::ContentTypeHeader,
+			  "application/json");
+
+	// GET is made async
+	settingsReply_ = restclient_->get(request);
+	// This is the callback when data is ready
+	connect(restclient_, &QNetworkAccessManager::finished, this,
+		&FacebookEncoderSettingsProvider::handleResponse);
+}
+
+QUrlQuery FacebookEncoderSettingsProvider::inputVideoQueryFromCurrentSettings()
+{
+	// Get input settings, shorten name
+	EncoderSettingsRequest *input = currentSettings_.get();
+
+	QUrlQuery inputVideoSettingsQuery;
+	inputVideoSettingsQuery.addQueryItem("video_type", "live");
+	if (input->userSelectedResolution.isNull()) {
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_width",
+			QString::number(input->videoWidth));
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_height",
+			QString::number(input->videoHeight));
+	} else {
+		QSize wizardResolution = input->userSelectedResolution.toSize();
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_width",
+			QString::number(wizardResolution.width()));
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_height",
+			QString::number(wizardResolution.height()));
+	}
+	inputVideoSettingsQuery.addQueryItem("input_video_framerate",
+					     QString::number(input->framerate));
+	inputVideoSettingsQuery.addQueryItem(
+		"input_video_bitrate", QString::number(input->videoBitrate));
+	inputVideoSettingsQuery.addQueryItem(
+		"input_audio_channels", QString::number(input->audioChannels));
+	inputVideoSettingsQuery.addQueryItem(
+		"input_audio_samplerate",
+		QString::number(input->audioSamplerate));
+	if (input->protocol == StreamProtocol::rtmps) {
+		inputVideoSettingsQuery.addQueryItem("cap_streaming_protocols",
+						     "rtmps");
+	}
+	// Defaults in OBS
+	inputVideoSettingsQuery.addQueryItem("cap_video_codecs", "h264");
+	inputVideoSettingsQuery.addQueryItem("cap_audio_codecs", "aac");
+	return inputVideoSettingsQuery;
+}
+
+QString FacebookEncoderSettingsProvider::getOBSVersionString()
+{
+	QString versionString;
+
+#ifdef HAVE_OBSCONFIG_H
+	versionString += OBS_VERSION;
+#else
+	versionString += LIBOBS_API_MAJOR_VER + "." + LIBOBS_API_MINOR_VER +
+			 "." + LIBOBS_API_PATCH_VER;
+#endif
+
+	return versionString;
+}
+
+// Helper methods for FacebookEncoderSettingsProvider::handleResponse
+void addInt(const QJsonObject &json, const char *jsonKey, SettingsMap *map,
+	    const char *mapKey)
+{
+	if (json[jsonKey].isDouble()) {
+		map->insert(mapKey,
+			    QPair(QVariant(json[jsonKey].toInt()), true));
+	}
+}
+
+void addStringDouble(const QJsonObject &json, const char *jsonKey,
+		     SettingsMap *map, const char *mapKey)
+{
+	if (!json[jsonKey].isString()) {
+		return;
```

Comment:
Is silent failure really OK in these helpers?

Reply:
Error handling and error UI isn't finished: see notes in " Work in Progress"